store/index-refactor.js

```diff
@@ -175,6 +183,21 @@ export const mutations = {
         p_state.currentPage = p_pageName;
     },
 
+    setDataDictionary(p_state, p_newDataDictionary, p_storeColumns) {
+
+        // Update values to existing columns in the data dictionary, but ignore any new columns
+        for ( const column of p_storeColumns ) {
+
+            if ( column in p_newDataDictionary ) {
+
+                for ( const key in p_newDataDictionary[column] ) {
```

Comment:
Is there a reason you're going through the keys in the column one by one here rather than just doing

`p_state.dataDictionary.annotated[column] = Object.assign({}, p_state.dataDictionary.annotated[column], p_newDataDictionary[column])`?

We've been using Object.assing so far for this kind of merging, and it'd be nice to be consistent. So if it is functionally equivalent, we should stick with Object.assign. If `Vue.set` does something `Object.assign` does not do, then we should use `Vue.set` instead.

Reply:
I think maybe you mean, 'Why are you going through key by key?' here? Given the clarified logic it does make sense to assign columns from the `newDataDictionary` but only after confirming that the column exists in the current data dictionary in the store.